### PR TITLE
Convert campaigns list to JSX

### PR DESCRIPTION
**Pull request by TheTrio**
This PR attempts to convert all campaigns list to React components. 

The list(or rather table) of campaigns is rendered on 3 routes. All of these routes have a different "version" of the list

1. `/explore` -> render only the 10 newest campaigns. Displays campaign statistics
2. `/` -> render the campaigns created by the user. Displays campaign statistics
3. `/campaigns` -> render all the campaigns. Display only the title and links to the CSVs. Has a search bar.

The views responsible for these were `/app/views/campaigns/_campaigns.html.haml` and `/app/views/campaigns/_row.html.haml`. They are no longer required and have been removed.

I've created several new components, most important of which is `CampaignList`. I've tried to make this component as generic as possible to allow sharing code between the lists rendered on the different routes. 

```js
const CampaignList = ({ keys, showSearch, RowElement, headerText, userOnly, newest }) => ...
```

I've also rewritten most of [`app/assets/javascripts/campaigns.js`](https://github.com/WikiEducationFoundation/WikiEduDashboard/pull/5026/commits/94e72eea21ae2773bfd2fbab6cfdc2d30ae8d43b#diff-782d559cd700c376e62d9282ff694ecf9372160ee8c7ef47ac94fd67a65202ea) to use vanilla JS instead of jQuery. Because of the use of the Editable jQuery plugin for editing campaigns, converting the rest of the file would require a lot more work - likely converting the whole campaigns route to a React component.

### CODE REVIEW

**Comment by ragesoss** [comment]
Please include before/after screenshots (or videos).

File: app/assets/javascripts/actions/campaign_actions.js [hunk 1]
```diff
@@ -93,9 +93,9 @@ const fetchAllCampaignsPromise = async () => {
   return response.json();
 };
 
-export const fetchAllCampaigns = () => (dispatch) => {
+export const fetchAllCampaigns = (userOnly, newest) => (dispatch) => {
```
Comment (ragesoss) [inline app/assets/javascripts/actions/campaign_actions.js:96]: Are these both supposed to be true/false values? A comment on how to use the parameters would be helpful. Does the name `fetchAllCampaigns` still accurately describe the function, or should it be renamed?

**Comment by TheTrio** [comment]
##  Campaigns Before

https://user-images.githubusercontent.com/10794178/177626791-581a5921-1807-4152-b0ab-ca6c68352e4b.mp4

## Campaigns After

https://user-images.githubusercontent.com/10794178/177626783-99e7be45-630d-4459-883f-0af8ea48bd12.mp4

## Dashboard Before

https://user-images.githubusercontent.com/10794178/177626807-481e13c4-12c3-4c02-936d-49fa69218683.mp4

## Dashboard After

https://user-images.githubusercontent.com/10794178/177626801-38a4cb10-58ae-4508-b2ca-35f85df8d9f0.mp4

## Explore Before

https://user-images.githubusercontent.com/10794178/177626816-76d8fb4e-6708-47ec-85ea-9bfdcbfd0d74.mp4

## Explore After

https://user-images.githubusercontent.com/10794178/177626810-5e709b6c-95c7-4002-b30f-050b729d4dcf.mp4

**Comment by ragesoss** [comment]
This looks very good! I will test it locally soon, but code-wise it looks clean and well-organized, and the videos look great.

File: app/assets/javascripts/actions/campaign_actions.js (hunk 1 above)
Comment (ragesoss) [reply app/assets/javascripts/actions/campaign_actions.js]: What  does `fetchCampaigns` do, and can it be renamed?

File: app/assets/stylesheets/modules/_overview.styl [hunk 2]
```diff
@@ -41,9 +41,6 @@
     color: green
     padding-top: 5px
 
-.tooltip-trigger
```
[the remark this answers is not in the document]
Comment (ragesoss) [reply app/assets/stylesheets/modules/_overview.styl:44]: Mobile is very bad overall these days, so I don't think there's much benefit in narrow things like setting 'desktop_only'. At some point — maybe a future internship project — we'll go more systematically through to get things tidy on mobile.

File: app/assets/javascripts/actions/campaign_actions.js (hunk 1 above)
Comment (TheTrio) [reply app/assets/javascripts/actions/campaign_actions.js]: I've renamed this to `fetchCampaignStatistics`. The `fetchAllCampaigns` function now returns all the campaigns(but without the statistics)

File: app/controllers/campaigns_controller.rb [hunk 3]
```diff
@@ -176,6 +170,14 @@ def remove_course
     redirect_to programs_campaign_path(@campaign.slug)
   end
 
+  def statistics
+    user_only = params[:user_only]
+    newest = params[:newest]
+    @values = user_only == 'true' ? current_user.campaigns : Campaign.all.order(created_at: :desc)
```
Comment (ragesoss) [inline app/controllers/campaigns_controller.rb:176]: This is not a good variable name. `@campaigns` instead?

**Comment by ragesoss** [comment]
This works smoothly. I'm ready to merge this after the redux action names are improved.

**Pull request merged**